xplique/metrics/fidelity.py

```diff
@@ -9,7 +9,7 @@
 from scipy.stats import spearmanr
 
 from .base import ExplanationMetric
-from ..commons import batch_predictions_one_hot
+from ..commons import batch_predictions
```

Comment:
In the whole fidelity file, `batch_predictions` is used, however, it corresponds to the default classification operator. In this sense, metric would be limited to classification.

I suggest that metric also take `operator` as an `__init__` argument and share the same code as BlackBoxExplainer in their init to define the `self.batch_predictions`. A quick patch could be to instanciate a dummy explainer and use its arguments.

Nonetheless, I think that in the long term, that all insertion metrics are merge into one supporting several data types and task. (Same for deletion).